code/modules/clothing/suits/plate_carrier.dm

```diff
@@ -0,0 +1,112 @@
+/**
+	Plate Carrier armor.
+		Armor that accepts an armored plate, that takes the brunt of the damage and steadily ablates to nothing.
+*/
+
+/obj/item/clothing/suit/armor/plate_carrier
+	name = "tactical plate armor"
+	icon_state = "tactical_armor"
+	item_state = "tactical_armor"
+	var/event_key
+	var/obj/item/weapon/armor_plate/P
+
+/obj/item/clothing/suit/armor/plate_carrier/get_armor(var/type)
+	var/armor_value = armor[type]
+	if(P)
+		armor_value = armor[type] <= 0 ? P.armor[type] : Clamp((armor[type]+P.armor[type])/2, armor[type], 100)
+	return armor_value
+
+/obj/item/clothing/suit/armor/plate_carrier/get_armor_absorb(var/type)
+	var/armor_value = armor_absorb[type]
+	if(P)
+		armor_value = armor_absorb[type] <= 0 ? P.armor_absorb[type] : Clamp((armor_absorb[type]+P.armor_absorb[type])/2, armor_absorb[type], 100)
+	return armor_value
+
+/obj/item/clothing/suit/armor/plate_carrier/equipped(var/mob/user, var/slot)
+	..()
+	if(slot == slot_wear_suit)
+		event_key = user.on_damaged.Add(src, "handle_user_damage")
+
+
+/obj/item/clothing/suit/armor/plate_carrier/unequipped(mob/user, var/from_slot = null)
+	if(from_slot == slot_wear_suit)
+		user.on_damaged.Remove(event_key)
+		event_key = null
+	..()
+
+/obj/item/clothing/suit/armor/plate_carrier/attack_self(mob/user)
+	if(P)
+		user.put_in_hands(P)
+		P = null
+
+/obj/item/clothing/suit/armor/plate_carrier/attackby(obj/item/W,mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/armor_plate) && !P && user.drop_item(W, src))
+		P = W
+		to_chat(user, "<span class = 'notice'>You install \the [W] into \the [src].</span>")
+
+/obj/item/clothing/suit/armor/plate_carrier/proc/handle_user_damage(list/arguments)
+	var/amount = arguments["amount"]
+	if(amount <= 0)
+		return
+	var/type = arguments["type"]
+
+	if(!P)
+		return
+	P.receive_damage(type, amount)
+	if(P.gcDestroyed)
+		P = null
+
+/obj/item/clothing/suit/armor/plate_carrier/security
+	name = "security plate armor"
+	icon_state = "security_armor"
+	item_state = "security_armor"
+	armor = list(melee = 10, bullet = 15, laser = 25, energy = 15, bomb = 5, bio = 0, rad = 0)
+
+/obj/item/weapon/armor_plate
+	icon = 'icons/obj/items.dmi'
+	icon_state = "plate_1"
+	name = "ceramic armor plate"
+	health = 20
+	armor = list(melee = 25, bullet = 7, laser = 50, energy = 10, bomb = 25, bio = 0, rad = 0)
+	armor_absorb = list(melee = 25, bullet = 20, laser = 20, energy = -5, bomb = 0, bio = 0, rad = 0)
+
+
+/obj/item/weapon/armor_plate/proc/receive_damage(var/type, var/amount)
+	if(type == BRUTE || type == BURN)
+		health -= amount
+	playsound(get_turf(src), 'sound/effects/Glasshit.ogg', 70, 1)
```

Comment:
I was able to get this to loop.  The mob was knocked over by ZAS and slammed into objects.  The plate was destroyed but the sound kept looping until I took the armor off the mob.  I'm not sure what this indicates.